tests/test_deprecated.py

```diff
@@ -137,14 +138,25 @@ def _func() -> int:
     assert decorated_func.__name__ == _func.__name__
     assert decorated_func.__doc__ == expected_func_doc
 
+    with pytest.warns(FutureWarning) as record:
+        decorated_func()
+    assert isinstance(record.list[0].message, Warning)
+    expected_warning_message = _deprecated._DEPRECATION_WARNING_TEMPLATE.format(
+        name=(name if name is not None else "_func"), d_ver="1.1.0", r_ver="3.0.0"
+    )
+    if text is not None:
+        expected_warning_message += " " + text
```

Comment:
Hmm, maybe I'm too conservative, but this is a copy of message generation logic in `optuna/_deprecated.py`, and I'm not sure if this can test if the logic actually generates the expected strings.

How about simplifying the expected values using `actual.endswith(expected_keyword)`, `expected_keyword in actual`

Reply:
If this test is passed, `actual.endswith(expected_keyword)` and `expected_keyword in actual` are also passed regardless of `_deprecated.py`. Do you mean to relax the constraint of the test?

Reply:
> If this test is passed, actual.endswith(expected_keyword) and expected_keyword in actual are also passed

I agree. But, I thinking we can avoid/reduce the conditional branch during the expected message generation. How about removing the parametrization of `name` since this test is for the `test` argument.

Reply:
OK, I reverted the last commit.

Reply:
Thanks!